src/state/mod.rs

```diff
@@ -236,9 +237,117 @@ impl IndexerState {
         })
     }
 
-    /// Creates a new indexer state from a db instance
-    pub fn new_from_db(database_dir: &Path) -> anyhow::Result<Self> {
-        todo!("Restoring from db in {}", database_dir.display());
+    pub fn new_with_db(
+        root_block: Block,
+        transition_frontier_length: Option<u32>,
+        store: IndexerStore,
+    ) -> anyhow::Result<Self> {
+        let root_branch =
+            Branch::new_non_genesis(root_block.state_hash.clone(), root_block.blockchain_length);
+        let tip = Tip {
+            state_hash: root_block.state_hash.clone(),
+            node_id: root_branch.root.clone(),
+        };
+
+        Ok(Self {
+            mode: IndexerMode::Light,
+            phase: IndexerPhase::Watching,
```

Comment:
There’s an `InitializingFromDB` phase for this

Reply:
thanks bestie got it